scripts/not-needed.cjs

```diff
@@ -18,10 +18,7 @@ rmdirRecursive(path.join('types', typingsPackageName));
 /**  @type  {{packages: Record<string, { libraryName: string; asOfVersion: string; }> }} */
 const notNeededPackages = JSON.parse(fs.readFileSync('notNeededPackages.json', 'utf-8'));
 notNeededPackages.packages[typingsPackageName] = { libraryName, asOfVersion };
-const sortedPackages = Object.entries(notNeededPackages.packages).sort((packageA, packageB) =>
-    packageA[0].localeCompare(packageB[0]),
-);
-notNeededPackages.packages = Object.fromEntries(sortedPackages);
+notNeededPackages.packages = Object.fromEntries(Object.entries(notNeededPackages.packages).sort());
```

Comment:
Is this correct? The entries are arrays, so this is going to stringify them when sorting, so depends on what arrays happen to stringify to.

Maybe extract out the keys, sort those, then recreate the object in order? Like:

```ts
Object.fromEntries(Object.keys(notNeededPackages.packages).sort().map(k => [k, notNeededPackages.packages[k]]))
```

Reply:
@jakebailey It's correct, I think. They're equivalent, given that:
1. No characters sort earlier than `,` (comma).
2. There aren't any duplicates.

1 is a property of npm package names:
- https://docs.npmjs.com/cli/v8/configuring-npm/package-json#name
- https://www.rfc-editor.org/rfc/rfc3986.html#section-2.3

2 is a property of JS object keys.

Reply:
I get what you mean, I just think it's a little dubious to depend on the output of `toString` for arrays. But, reading the ECMAScript spec, Array's `toString` is defined as calling `join`, so, I guess it's safe since the values are going to always stringify to `[object Object]`, just a little slower than sorting only the keys themselves.